spec/steps/provision.fmf

```diff
@@ -97,3 +96,47 @@ example: |
             how: connect
             guest: hostname or ip address
             key: private-key-path
+
+/environment:
+    summary: Environment specification
+
+    description: |
+        As part of the ``provision`` step it is possible to
+        specify additional requirements for the testing
+        environment such as required amount of ``memory`` and
+        ``disk`` space.
+
+        FIXME: Much more details go here. For now let's start with
+        a couple of tangible examples.
+
+    example: |
+        # Simple use cases simple
+        memory: 8 GB
+        disk: 500 GB
+
+        # Processor-related stuff grouped together
+        cpu:
+            processors: 2
+            cores: 16
+            model: 37
+
+        # Operators at the start of the value
+        memory: '> 8 GB'
+        memory: '>= 8 GB'
+        memory: '< 8 GB'
+        memory: '<= 8 GB'
+
+        # By default exact value expected, these are equivalent:
```

Comment:
these all should go under `cpu`

Reply:
Fixed. But looking at that I'm thiking whether the nesting/grouping similar options under a single parent is a good idea. What benefits does it bring? Wouldn't be better to have `cpu_processors`, `cpu_cores` and `cpu_model` instead?

Reply:
At least it wouldn't misuse property names to express hierarchy: - instead of `cpu_cores`, `cpu_model` and so on, I can grab `cpu` and *know* cpu-related bits are in this container. I don't have to check all properties on top-level, whether they by any chance start with `cpu_` prefix. My 2 cents.

Reply:
Yeah, let's stick with the hierarchy. Seems as a better approach.